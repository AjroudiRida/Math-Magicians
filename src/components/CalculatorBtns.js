import PropTypes from 'prop-types';

const CalculatorBtns = ({ content }) => (
  <div className="btn">
    <span>{content}</span>
  </div>
);

CalculatorBtns.propTypes = {
  content: PropTypes.string.isRequired,
};

export default CalculatorBtns;
