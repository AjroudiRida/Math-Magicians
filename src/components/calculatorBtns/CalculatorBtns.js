import PropTypes from 'prop-types';
import './buttons.css';

const CalculatorBtns = ({ content, handleClick }) => (
  <button
    type="button"
    className="btn"
    onClick={handleClick}
    onKeyDown={handleClick}
    data-testid="button"
  >
    <span>{content}</span>
  </button>
);
CalculatorBtns.propTypes = {
  content: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired,
};
export default CalculatorBtns;
