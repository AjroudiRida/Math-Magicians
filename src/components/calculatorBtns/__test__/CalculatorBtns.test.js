import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import CalculatorBtns from '../CalculatorBtns';

const dummyFunction = () => {};

describe('calculator btns tests', () => {
  it('should render in the DOM', () => {
    render(<CalculatorBtns content="+" handleClick={dummyFunction} />);
    const btnElement = screen.getByText('+');
    expect(btnElement).toBeInTheDocument();
  });

  it('should be visible in the DOM', () => {
    render(<CalculatorBtns content="+" handleClick={dummyFunction} />);
    const btnElement = screen.getByText('+');
    expect(btnElement).toBeVisible();
  });

  it('should display text passed as prop', () => {
    render(<CalculatorBtns content="2" handleClick={dummyFunction} />);
    const btnElement = screen.getByText(/2/i);
    expect(btnElement).toHaveTextContent('2');
  });

  it('should match the Dom snapshot', () => {
    const btnsTree = renderer
      .create(<CalculatorBtns content="+" handleClick={dummyFunction} />)
      .toJSON();
    expect(btnsTree).toMatchSnapshot();
  });
});
