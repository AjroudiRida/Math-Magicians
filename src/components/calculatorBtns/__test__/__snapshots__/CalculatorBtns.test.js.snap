// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculator btns tests should match the Dom snapshot 1`] = `
<button
  className="btn"
  data-testid="button"
  onClick={[Function]}
  onKeyDown={[Function]}
  type="button"
>
  <span>
    +
  </span>
</button>
`;
